mock/api.js

```diff
@@ -1,381 +1,297 @@
-              measurement_idx: 0,
-              measurement_title: 'test_measurement_title',
-              '@idx': 0,
-              name: 'sample1',
-            },
-          },
-        },
-        {
-          _source: {
-            run: {
-              id: 'test_run_id',
-              controller: 'test_controller',
-              name: 'test_run_name',
-              script: 'test_script',
-              config: 'test_config',
-            },
-            iteration: { name: 'test_iteration_2', number: 2 },
-            benchmark: {
-              instances: 1,
-              max_stddevpct: 1,
-              message_size_bytes: 1,
-              primary_metric: 'test_measurement_title',
-              test_type: 'stream',
-            },
-            sample: {
-              closest_sample: 1,
-              mean: 0.1,
-              stddev: 0.1,
-              stddevpct: 1,
-              uid: 'test_measurement_id',
-              measurement_type: 'test_measurement_type',
-              measurement_idx: 0,
-              measurement_title: 'diff_measurement_title',
-              '@idx': 1,
-              name: 'sample2',
-            },
-          },
-        },
-      ],
-    },
-    aggregations: {
-      id: {
-        buckets: [
-          {
-            key: 'test_run_name',
-            type: {
-              buckets: [
-                {
-                  key: 'test_measurement_type',
-                  title: {
-                    buckets: [
-                      {
-                        key: 'test_measurement_title',
-                        uid: {
-                          buckets: [
-                            {
-                              key: 'test_measurement_id',
-                            },
-                          ],
-                        },
-                      },
-                    ],
-                  },
-                },
-              ],
-            },
-          },
+export const mockTableContents = {
+  hits: {
+    total: DEFAULT_SIZE,
+    hits: new Array(DEFAULT_SIZE).fill().map((value, index) => ({
+      _source: {
+        parent: `${casual.word}/${casual.word}/${casual.word}/${casual.word}/${casual.word}/${
+          casual.word
+        }`,
+        directory: `${casual.word}/${casual.word}/${casual.word}/${casual.word}/${casual.word}/${
+          casual.word
+        }`,
+        mtime: moment.utc() + index,
+        mode: casual.word,
+        name: casual.word,
+        ancestor_path_elements: [
+          casual.word,
+          casual.word,
+          casual.word,
+          casual.word,
+          casual.word,
+          casual.word,
         ],
-      },
-      name: {
-        buckets: [
+        files: [
           {
-            key: 'test_run_name',
+            name: casual.uid,
+            mtime: moment.utc() + index,
+            size: casual.integer(1, DEFAULT_SIZE),
+            mode: casual.word,
+            type: casual.word,
```

Comment:
Similar comment here: should we worry about dashboard code that cares about the file type "keywords" returned from Elasticsearch and be more careful here about mocking? Maybe just hardcoded `"reg"` would be good enough?

Reply:
Thanks for pointing this out! Is there a definition for type of "keywords" we can expect from Elasticsearch? In that case, we can include a solution that randomly selects a keyword type from the list for each mocked object.

Reply:
It's a lookup from the tar file type codes, in `indexer.py`:

```
    _mode_table = _dict_const(
        [
            (tarfile.REGTYPE, "reg"),
            (tarfile.AREGTYPE, "areg"),
            (tarfile.LNKTYPE, "lnk"),
            (tarfile.SYMTYPE, "sym"),
            (tarfile.DIRTYPE, "dir"),
            (tarfile.FIFOTYPE, "fifo"),
            (tarfile.CONTTYPE, "cont"),
            (tarfile.CHRTYPE, "chr"),
            (tarfile.BLKTYPE, "blk"),
            (tarfile.GNUTYPE_SPARSE, "spr"),
        ]
    )
```

Reply:
I've updated the mock generator for the `type` field here to randomly select a tar file type upon generation.